### PR TITLE
Make gen_pipeline.py functional for 6X_STABLE branch

### PR DESCRIPTION
**Pull request by oliverralbertini**
w/ @bradfordboyle 
Our README in `gp-continuous-integration` correctly states how to generate the 6X_STABLE pipeline, however the gen_pipeline.py tool wasn't doing the right thing.

This PR addresses that and improves the tool based off of some suggestions from a py linter.

### CODE REVIEW

**Comment by frankgh** [comment]
I'm referring to these fly commands: https://github.com/greenplum-db/gpdb/blame/6X_STABLE/concourse/pipelines/README.md#L87

File: concourse/pipelines/gen_pipeline.py [hunk 1]
```diff
@@ -57,30 +57,31 @@
...
+                                      'gpdb_packaging_ubuntu16',
+                                      'resource_group_sles12',
+                                      'madlib_build_gppkg',
+                                      'MADlib_Test_planner_centos6',
+                                      'MADlib_Test_orca_centos6',
+                                      'MADlib_Test_planner_centos7',
+                                      'MADlib_Test_orca_centos7',
+                                      'icw_extensions_gpcloud_ubuntu16'] +
+                                      RELEASE_VALIDATOR_JOB +
+                                      JOBS_THAT_ARE_GATES)
```
Comment (benchristel) [inline concourse/pipelines/gen_pipeline.py]: I'd prefer not to change the whitespace here. Having everything aligned to the right will just lead to future spurious diffs if someone renames `JOBS_THAT_SHOULD_NOT_BLOCK_RELEASE`.

File: concourse/pipelines/README.md [hunk 2]
```diff
@@ -71,35 +71,37 @@ $ ./gen_pipeline.py -t prod
...
     -l ~/workspace/gp-continuous-integration/secrets/gpdb_common-ci-secrets.yml \
-    -l ~/workspace/gp-continuous-integration/secrets/gpdb_master_without_asserts-ci-secrets.yml
+    -l ~/workspace/gp-continuous-integration/secrets/gpdb_6X_STABLE_without_asserts-ci-secrets.prod.yml \
+    -v pipeline-name=6X_STABLE_without_asserts
 ```
 
-The generated pipeline file `gpdb_master-generated.yml` will be set,
+The generated pipeline file `gpdb_6X_STABLE-generated.yml` will be set,
 validated and ultimately committed (including the updated pipeline
 template) to the source repository.
```
Comment (benchristel) [inline concourse/pipelines/README.md]: Not the fault of this PR, but why do we duplicate the output of `gen_pipeline.py` in this document? Why not just let the tooling speak for itself?

File: concourse/pipelines/gen_pipeline.py [hunk 3]
```diff
@@ -268,17 +286,26 @@ def how_to_use_generated_pipeline_message():
...
+                              'UD',
+                              'AA',
+                              'Extensions',
+                              'Gpperfmon']
+
+    # if generating a dev pipeline but didn't specify an output,
+    # don't overwrite the 6X_STABLE pipeline
+    if (ARGS.pipeline_type != 'prod' and
+            os.path.basename(ARGS.output_filepath) == DEFAULT_OUTPUT_FILENAME):
+        DEFAULT_DEV_OUTPUT_FILENAME = 'gpdb-' + ARGS.pipeline_type + '-' + ARGS.user + '.yml'
```
Comment (benchristel) [inline concourse/pipelines/gen_pipeline.py]: Python newbie question: Why is `DEFAULT_DEV_OUTPUT_FILENAME` uppercase now? Is that a Python convention for something?

Comment (bradfordboyle) [reply concourse/pipelines/gen_pipeline.py]: The `flake8` linter prints the following error/warning:

> invalid-name: Constant name "default_dev_output_filename" doesn't conform to UPPER_CASE naming style

**Comment by bradfordboyle** [comment]
I've cleaned up this PR to separate out the lint clean up and opened #7339 targeting master. The goal is to make this PR smaller and faster to merge to 6X_STABLE.

**Comment by bradfordboyle** [comment]
@benchristel || @jchampio || @dkrieger-pvtl 

Let me know if this is OK to merge to 6X_STABLE.

**Comment by dkrieger-pvtl** [comment]
I checked out the gen_pipeline.py file onto my development branch off of 6X_STABLE and am able to successfully generate the 6X_STABLE and 6X_stable_without_asserts.

You'll need to rebase from 6X_STABLE and regenerate the main yml file, as my last commit changed it significantly.  

To me, the changes are fine to merge now, and the other comments can be addressed separately, as the main point of this fix--to make it easy to generate 6X_STABLE pipelines--is done.

**Pull request merged**